src/runtime/loader.ts

```diff
@@ -76,34 +76,42 @@ export function extractStack(
         entryPoint: funcName.replace(/-/g, '.'),
       };
       if (val['__requiredAPIs'] && Array.isArray(val['__requiredAPIs'])) {
-        for (const { api, reason } of val['__requiredAPIs']) {
-          const reasons = [reason];
-          if (requiredAPIs[api]?.reason) {
-            reasons.push(requiredAPIs[api].reason);
-          }
-          requiredAPIs[api] = { api, reason: reasons.join(' ') };
-        }
+        requiredAPIs.push(...val['__requiredAPIs']);
       }
     } else if (typeof val === 'object' && val !== null) {
       extractStack(val, endpoints, requiredAPIs, prefix + name + '-');
     }
   }
 }
 
+function mergeRequiredAPIs(
+  requiredAPIs: ManifestRequiredAPI[]
+): ManifestRequiredAPI[] {
+  const apiToReasons: Record<string, Set<string>> = {};
+  for (const { api, reason } of requiredAPIs) {
+    const reasons = apiToReasons[api] || new Set();
```

Comment:
I'm confused. Doesn't this never actually insert a new key into `apiToReasons`?

Reply:
🤦🏼 Definitely need tests here.